### PR TITLE
[MAINT] Format regions module

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates #2528

Changes proposed in this pull request:

- apply black and isort
- fix remaining flake8 issues
- remove dead code flagged by vulture
- apply some refactoring from sourcery
- fix f strings

### CODE REVIEW

File: nilearn/regions/hierarchical_kmeans_clustering.py [hunk 1]
```diff
@@ -202,24 +233,33 @@ def fit(self, X, y=None):
...
+            X, ensure_min_features=2, ensure_min_samples=2, estimator=self
+        )
         n_features = X.shape[1]
 
         if self.n_clusters <= 0:
-            raise ValueError("n_clusters should be an integer greater than 0."
-                             " %s was provided." % str(self.n_clusters))
+            raise ValueError(
+                "n_clusters should be an integer greater than 0."
+                " %s was provided." % str(self.n_clusters)
```
Comment (Remi-Gau) [inline nilearn/regions/hierarchical_kmeans_clustering.py:244]: TODO: format f strings

File: nilearn/regions/hierarchical_kmeans_clustering.py [hunk 2]
```diff
@@ -202,24 +233,33 @@ def fit(self, X, y=None):
...
-            warnings.warn("n_clusters should be at most the number of "
-                          "features. Taking n_clusters = %s instead."
-                          % str(n_features))
-        self.labels_ = hierarchical_k_means(X, self.n_clusters, self.init,
-                                            self.batch_size, self.n_init,
-                                            self.max_no_improvement,
-                                            self.verbose, self.random_state)
+            warnings.warn(
+                "n_clusters should be at most the number of "
+                "features. Taking n_clusters = %s instead." % str(n_features)
```
Comment (Remi-Gau) [inline nilearn/regions/hierarchical_kmeans_clustering.py:251]: TODO: format f strings

File: nilearn/regions/parcellations.py [hunk 3]
```diff
@@ -62,15 +62,13 @@ def _estimator_fit(data, estimator, method=None):
...
-    """
+    """Check the parameters and prepare for processing as a list."""
     imgs = stringify_path(imgs)
     confounds = stringify_path(confounds)
-    if not isinstance(imgs, (list, tuple)) or \
-            isinstance(imgs, str):
-        imgs = [imgs, ]
+    if not isinstance(imgs, (list, tuple)) or isinstance(imgs, str):
+        imgs = [
+            imgs,
```
Comment (Remi-Gau) [inline nilearn/regions/parcellations.py:70]: remove comma

File: nilearn/regions/parcellations.py [hunk 4]
```diff
@@ -80,20 +78,23 @@ def _check_parameters_transform(imgs, confounds):
...
 
-    if confounds is not None:
-        if not isinstance(confounds, (list, tuple)) or \
-                isinstance(confounds, str):
-            confounds = [confounds, ]
+    if confounds is not None and (
+        not isinstance(confounds, (list, tuple)) or isinstance(confounds, str)
+    ):
+        confounds = [
+            confounds,
```
Comment (Remi-Gau) [inline nilearn/regions/parcellations.py:85]: remove comma

File: nilearn/regions/parcellations.py [hunk 5]
```diff
@@ -329,60 +359,77 @@ def _raw_fit(self, data):
         # plotting to avoid backend display error with matplotlib
         # happening in Travis
         try:
-            from nilearn import plotting
+            from nilearn import plotting  # noqa
```
Comment (Remi-Gau) [inline nilearn/regions/parcellations.py:362]: flake8 says this plotting is unused

silencing for now

File: nilearn/regions/tests/test_region_extractor.py [hunk 6]
```diff
@@ -289,51 +314,62 @@ def test_connected_label_regions():
...
+    extract_reg_min_size_large = connected_label_regions(
+        labels_img, min_size=500
+    )
     assert np.unique(get_data(extract_reg_min_size_large)) == 0
 
     # Test the names of the brain regions given in labels.
     # Test labels for 9 regions in n_regions
-    labels = ['region_a', 'region_b', 'region_c', 'region_d', 'region_e',
-              'region_f', 'region_g', 'region_h', 'region_i']
+    labels = [
```
Comment (NicolasGensollen) [inline nilearn/regions/tests/test_region_extractor.py:342]:
```python
labels =  [f"region_{x}" for x in "abcdefghi"]
```

File: nilearn/regions/tests/test_region_extractor.py [hunk 7]
```diff
@@ -345,8 +381,14 @@ def test_connected_label_regions():
     unique_labels.remove(0)
 
     # labels given are less than n_regions=9
-    provided_labels = ['region_a', 'region_c', 'region_f',
-                       'region_g', 'region_h', 'region_i']
+    provided_labels = [
```
Comment (NicolasGensollen) [inline nilearn/regions/tests/test_region_extractor.py:384]:
```python
provided_labels =  [f"region_{x}" for x in "acfghi"]
```

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @Remi-Gau !
LGTM with a few suggestions.

File: nilearn/regions/tests/test_region_extractor.py (hunk 6 above)
Comment (Remi-Gau) [reply nilearn/regions/tests/test_region_extractor.py]: nice one.

implemented

File: nilearn/regions/tests/test_region_extractor.py (hunk 7 above)
Comment (Remi-Gau) [reply nilearn/regions/tests/test_region_extractor.py]: nice one.

implemented

File: nilearn/regions/rena_clustering.py [hunk 8]
```diff
@@ -485,41 +516,54 @@ def fit(self, X, y=None):
...
+        )
         n_features = X.shape[1]
 
         if not isinstance(self.mask_img, (str, Nifti1Image)):
-            raise ValueError("The mask image should be a Niimg-like"
-                             "object. Instead a %s object was provided."
-                             % type(self.mask_img))
+            raise ValueError(
+                "The mask image should be a Niimg-like"
+                "object. Instead a %s object was provided."
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/regions/rena_clustering.py]: done

File: nilearn/regions/rena_clustering.py [hunk 9]
```diff
@@ -394,17 +413,18 @@ def recursive_neighbor_agglomeration(X, mask_img, n_clusters,
...
 
         labels = reduced_labels[labels]
         n_components = connectivity.shape[0]
 
         if verbose > 0:
-            print('After iteration number %s, features are '
-                  ' grouped into %s clusters'
-                  % (str(i + 1), str(n_components)))
+            print(
+                "After iteration number %s, features are "
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/regions/rena_clustering.py]: done

File: nilearn/regions/region_extractor.py [hunk 10]
```diff
@@ -483,45 +535,51 @@ def connected_label_regions(labels_img, min_size=None, connect_diag=True,
...
 
     if labels is not None:
-        if (not isinstance(labels, collections.abc.Iterable) or
-                isinstance(labels, str)):
-            labels = [labels, ]
+        if not isinstance(labels, collections.abc.Iterable) or isinstance(
+            labels, str
+        ):
+            labels = [
+                labels,
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/regions/region_extractor.py]: good catch

File: nilearn/regions/parcellations.py [hunk 11]
```diff
@@ -521,21 +574,23 @@ def inverse_transform(self, signals):
...
         self._check_fitted()
 
-        if not isinstance(signals, (list, tuple)) or\
-                isinstance(signals, np.ndarray):
-            signals = [signals, ]
+        if not isinstance(signals, (list, tuple)) or isinstance(
+            signals, np.ndarray
+        ):
+            signals = [
+                signals,
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/regions/parcellations.py]: done

File: nilearn/regions/parcellations.py (hunk 5 above)
Comment (Remi-Gau) [reply nilearn/regions/parcellations.py]: Commented the import and let's see what CI says.

File: nilearn/regions/tests/test_parcellations.py [hunk 12]
```diff
@@ -171,32 +178,35 @@ def test_check_parameters_transform(test_image_2):
...
 
 
-@pytest.mark.parametrize('method', METHODS)
-@pytest.mark.parametrize('n_parcel', [5])
-def test_parcellations_transform_with_multi_confounds_multi_images(method, n_parcel, test_image_2):  # noqa: E501
+@pytest.mark.parametrize("method", METHODS)
+@pytest.mark.parametrize("n_parcel", [5])
+def test_parcellations_transform_with_multi_confounds_multi_images(
+    method, n_parcel, test_image_2
+):  # noqa: E501
```
Comment (ymzayek) [inline nilearn/regions/tests/test_parcellations.py:195]: Remove ignore error statement

File: nilearn/regions/tests/test_signal_extraction.py [hunk 13]
```diff
@@ -52,26 +52,28 @@ def test__check_shape_and_affine_compatibility_with_dim():
...
+        )
+
 
 @pytest.mark.parametrize(
-    "test_shape,test_affine,msg", 
-    [((2, 3, 5), np.eye(4), _TEST_SHAPE_ERROR_MSG),
-     ((2, 3, 4), 2*np.eye(4), _TEST_AFFINE_ERROR_MSG)
-     ])
-def test__check_shape_and_affine_compatibility_error(test_shape,test_affine,msg):
+    "test_shape,test_affine,msg",
```
Comment (ymzayek) [inline nilearn/regions/tests/test_signal_extraction.py:60]:
```suggestion
    "test_shape, test_affine, msg",
```

File: nilearn/regions/signal_extraction.py [hunk 14]
```diff
@@ -544,27 +545,30 @@ def _trim_maps(maps, mask, keep_empty=False, order="F"):
...
     if keep_empty:
-        return trimmed_maps, maps_mask, np.arange(trimmed_maps.shape[-1],
-                                                  dtype=int)
+        return (
+            trimmed_maps,
+            maps_mask,
+            np.arange(trimmed_maps.shape[-1], dtype=int),
+        )
     else:
         return trimmed_maps, maps_mask, np.where(sums > 0)[0]
```
Comment (ymzayek) [inline nilearn/regions/signal_extraction.py:574]: Maybe instead

```suggestion
    indices = np.arange(
        trimmed_maps.shape[-1], dtype=int
    ) if keep_empty else np.where(sums > 0)[0]

    return trimmed_maps, maps_mask, indices
```

File: nilearn/regions/parcellations.py (hunk 5 above)
Comment (Remi-Gau) [reply nilearn/regions/parcellations.py]: CI passes without this so I will remove it.

File: nilearn/regions/tests/test_signal_extraction.py (hunk 13 above)
Comment (Remi-Gau) [reply nilearn/regions/tests/test_signal_extraction.py:60]: Unsure about that one.
Although I don't prefer it, I think I have almost always seen parameters being separated by just a comma.

If we start doing this I may to recheck previously formatted module.

Also there is no way that I know of to enforce this.

Comment (ymzayek) [reply nilearn/regions/tests/test_signal_extraction.py:60]: Ok I didn't know that. Then no worries, leave it as is

File: nilearn/regions/tests/test_parcellations.py (hunk 12 above)
Comment (Remi-Gau) [reply nilearn/regions/tests/test_parcellations.py]: done

File: nilearn/regions/signal_extraction.py (hunk 14 above)
Comment (Remi-Gau) [reply nilearn/regions/signal_extraction.py]: done

**Review by bthirion — COMMENTED** [review]
LGTM overall.
We should later fix the coverage warnings.

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks @Remi-Gau !

**Comment by Remi-Gau** [comment]
thanks @NicolasGensollen and @bthirion 

merging

**Pull request merged**